### PR TITLE
Enhance test_solve.py and test_matrix_power.py

### PR DESCRIPTION
**Pull request by robinw0928**

### CODE REVIEW

File: tests/integration/test_matrix_power.py [hunk 1]
```diff
@@ -22,18 +22,88 @@
...
     res_np = np.linalg.matrix_power(a_np, exp)
     res_num = num.linalg.matrix_power(a_num, exp)
-    assert allclose(res_np, res_num)
+    assert np.array_equal(res_np, res_num)
+
+
+class TestMatrixPowerErrors:
+    @pytest.mark.parametrize("ndim", (0, 1), ids=lambda ndim: f"(ndim={ndim})")
+    def test_matrix_ndim_smaller_than_two(self, ndim):
+        # In Numpy, it raises numpy.linalg.LinAlgError
```
Comment (bryevdv) [inline tests/integration/test_matrix_power.py:74]: If numpy is raising `LinAlgError` that's definitely on purpose (i.e. we *do* definitely want to match exception *type* then) so `xfail` here (and anywhere else that misses raising `LinAlgError`)

File: tests/integration/test_solve.py [hunk 2]
```diff
@@ -80,6 +80,109 @@ def test_solve_corner_cases():
...
+    b = num.atleast_2d([])
+
+    out = num.linalg.solve(a, b)
+    assert np.array_equal(b, out)
+
+
+def test_solve_dtype_int():
+    n = 8
+    a = np.random.randint(0, 10, size=(n, n), dtype="i4")
+    b = np.random.randint(0, 10, size=(n,), dtype="i4")
```
Comment (bryevdv) [inline tests/integration/test_solve.py:94]: do these not work with 64 bit int types?

Comment (bryevdv) [reply tests/integration/test_solve.py:94]: BTW I do believe that there is a non-vanishing probability that a random matrix of non-negative integers is singular. Is there a reason to make these random?

File: tests/integration/test_solve.py [hunk 3]
```diff
@@ -80,6 +80,109 @@ def test_solve_corner_cases():
...
+
+
+class TestSolveErrors:
+    def setup_method(self):
+        self.n = 3
+        self.a = num.random.rand(self.n, self.n).astype("d")
+        self.b = num.random.rand(self.n).astype("d")
+
+    def test_a_bad_dim(self):
+        a = num.random.rand(self.n).astype("d")
```
Comment (bryevdv) [inline tests/integration/test_solve.py:130]: why is this random test value needed?

File: tests/integration/test_solve.py [hunk 4]
```diff
@@ -80,6 +80,109 @@ def test_solve_corner_cases():
...
+            num.linalg.solve(a, self.b)
+
+    def test_b_bad_dim(self):
+        b = 10
+        msg = "Array must be at least one-dimensional"
+        with pytest.raises(num.linalg.LinAlgError, match=msg):
+            num.linalg.solve(self.a, b)
+
+    def test_a_bad_dtype_float16(self):
+        a = self.a.astype("e")
```
Comment (bryevdv) [inline tests/integration/test_solve.py:147]: suggest using actual np dtype objects, or the more descriptive string names. AFAIK the one-letter codes are discouraged by the numpy devs at this point.

File: tests/integration/test_matrix_power.py (hunk 1 above)
Comment (robinw0928) [reply tests/integration/test_matrix_power.py]: I fixed the source code so that the tests pass now.

File: tests/integration/test_solve.py (hunk 2 above)
Comment (robinw0928) [reply tests/integration/test_solve.py]: I test both np.int32 and np.int64 now. 
You are right. There is probability that the random int matrix is singular. Fixed.

File: tests/integration/test_solve.py (hunk 4 above)
Comment (robinw0928) [reply tests/integration/test_solve.py]: Fixed.

File: tests/integration/test_solve.py (hunk 3 above)
Comment (robinw0928) [reply tests/integration/test_solve.py:131]: I think random here is a way of generating a one-dimension array with shape (self.n,). It works.

**Review by bryevdv — APPROVED** [review]

File: tests/integration/test_solve.py (hunk 4 above)
Comment (bryevdv) [reply tests/integration/test_solve.py]: FWIW there still are quite a few other instances in these files using the single-letter codes.

Comment (robinw0928) [reply tests/integration/test_solve.py]: Fixed. Thanks.

**Pull request merged**